{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.insert(0, \"/Users/hamzaharunamohammed/Desktop/ntnu/PhD/Development/streamlit_llm_models_deployment\")"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "outputs": [],
   "source": [
    "from implementing_rag.rag_chromadb_engine import RagChromaDbEngine"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [],
   "source": [
    "rag_chromadb_engine = RagChromaDbEngine()"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [],
   "source": [
    "prompt = \"From the univariable analyses comparing miR concentrations between patients\"\n",
    "context = rag_chromadb_engine.generate_context(query=prompt, n_results=3)"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [
    {
     "data": {
      "text/plain": "'Polygenic risk scores (PRSs) may provide new insights into the genetic basis behind the associations among lifestyles, disease risks and mortality. Individuallevel genetic risk estimates, that is, the PRSs, are generated by summarising genomewide SNPs and the associated effect sizes into a single variable. PRSs have been used to estimate an individual’s genetic propensity for multiple diseases and traits . Recently, PRSs of the lifetime risk for coronary heart diseases have been found to weakly improve prediction models for coronary heart disease compared with models based on traditional risk factors [ 13]. Sillanpää et al. [ 14] found that PRSs for PA (PA PRSs) were weakly associated with several noncommunicable diseases, suggesting small pleiotropic effects. Recently, in their multiancestry metaanalysis of genomewide association studies (GWASs) based on over 700,000 persons, Wang et al. [ 8] found that selfreported moderatetovigorous leisuretime PA (LTPA) correlated weakly or moderately with multiple anthropometric characteristics, lifestyle factors, noncommunicable diseases and biomarkers at the genomic level. A possible reason may be the difficulty of measuring PA accurately over sufficient time periods to obtain informative and reliable measures of lifetime physical activity\\nPolygenic risk scores (PRSs) may provide new insights into the genetic basis behind the associations among lifestyles, disease risks and mortality. Individuallevel genetic risk estimates, that is, the PRSs, are generated by summarising genomewide SNPs and the associated effect sizes into a single variable. PRSs have been used to estimate an individual’s genetic propensity for multiple diseases and traits . Recently, PRSs of the lifetime risk for coronary heart diseases have been found to weakly improve prediction models for coronary heart disease compared with models based on traditional risk factors [ 13]. Sillanpää et al. [ 14] found that PRSs for PA (PA PRSs) were weakly associated with several noncommunicable diseases, suggesting small pleiotropic effects. Recently, in their multiancestry metaanalysis of genomewide association studies (GWASs) based on over 700,000 persons, Wang et al. [ 8] found that selfreported moderatetovigorous leisuretime PA (LTPA) correlated weakly or moderately with multiple anthropometric characteristics, lifestyle factors, noncommunicable diseases and biomarkers at the genomic level. A possible reason may be the difficulty of measuring PA accurately over sufficient time periods to obtain informative and reliable measures of lifetime physical activity\\nPolygenic risk scoring We utilised previously derived PRS for the selfreported moderate PA [ 15]. In the original PanUK Biobank, GWAS moderate PA was determined based on the selfreport question on the ‘number of days/week of moderate PA 10 + min’ . To construct the PA PRS in the HUNT Study we utilised SBayesR summary statistics methodology in the GCBT software [ 22], and the PanUK Biobank GWAS Fig. 1 Design and workflow of the study. Polygenic score for questionnairebased moderate physical activity was derived from the PanUK Biobank genomewide association study summary statistics and the third cohort of the HUNT Study (HUNT3; N = 47,148). Association analyses were conducted in HUNT3 and its subcohorts. The cardiometabolic disease (CMD) endpoints were derived from a Norwegian hospital discharge register 1 3997 N. P . Tynkkynen et al.Tampa, USA). The arithmetic average of the second and third measurements was used. A combined scale (Model DS102, Nøtterøy, Norway) was used to measure weight (kg) and height (cm). BMI was calculated as weight divided by height squared (kg/m2). The waist circumference was measured horizontally at the umbicillus height (cm)'"
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "context"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
